### PR TITLE
[WIP-pr] Feature/cyber markets#100

### PR DESCRIPTION
**Pull request by mgergalov**
#100 issue

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -94,15 +96,87 @@ subprojects {
...
 }
 
-project(":connectors") {
+project(":connectors-core") {
     apply plugin: "application"
 
     dependencies {
         compile(project(":core"))
         compile("io.undertow:undertow-core")
+        compile ("org.web3j:core:$web3jVersion")
```
Comment (hleb-albau) [inline build.gradle]: web3j only for ehtereum, remove

File: build.gradle [hunk 2]
```diff
@@ -94,15 +96,87 @@ subprojects {
...
+    }
+}
+
+project(":connectors-bitfinex") {
+    apply plugin: "application"
+
+    dependencies {
+        compile(project(":core"))
+        compile(project(":connectors-core"))
+        compile("io.undertow:undertow-core")
```
Comment (hleb-albau) [inline build.gradle]: already in core

File: connectors-bitstamp/src/main/kotlin/fund/cyber/markets/connectors/bitstamp/BitstampConnectorApplication.kt [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import kotlinx.coroutines.experimental.newFixedThreadPoolContext
+import kotlinx.coroutines.experimental.newSingleThreadContext
+import okhttp3.Dispatcher
+import okhttp3.OkHttpClient
+import org.xnio.OptionMap
+import org.xnio.Options
+import org.xnio.Xnio
+import org.xnio.XnioWorker
+
+/*----------------------------- Undertow ------------------------------------------*/
```
Comment (hleb-albau) [inline connectors-bitstamp/src/main/kotlin/fund/cyber/markets/connectors/bitstamp/BitstampConnectorApplication.kt]: already in core

File: connectors-core/src/test/kotlin/fund/cyber/markets/exchanges/hitbtc/HitBtcMessageParserTest.kt [hunk 4]
```diff
@@ -2,8 +2,8 @@ package fund.cyber.markets.exchanges.hitbtc
 
 import fund.cyber.markets.connectors.common.ContainingUnknownTokensPairMessage
 import fund.cyber.markets.connectors.common.TradesUpdatesMessage
-import fund.cyber.markets.connectors.hitbtc.HitBtcTokensPairInitializer
-import fund.cyber.markets.connectors.hitbtc.HitBtcTradesMessageParser
+import fund.cyber.markets.connectors.etherdelta.connector.HitBtcTokensPairInitializer
```
Comment (hleb-albau) [inline connectors-core/src/test/kotlin/fund/cyber/markets/exchanges/hitbtc/HitBtcMessageParserTest.kt]: etherdelta package?

File: connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt [hunk 5]
```diff
@@ -1,17 +1,8 @@
-package fund.cyber.markets.connectors
+package fund.cyber.markets.connectors.etherdelta
```
Comment (hleb-albau) [inline connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt]: package

File: connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt [hunk 6]
```diff
@@ -25,10 +16,8 @@ import org.xnio.Options
 import org.xnio.Xnio
 import org.xnio.XnioWorker
 
-
 /*----------------------------- Undertow ------------------------------------------*/
```
Comment (hleb-albau) [inline connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt]: already in core

File: connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt [hunk 7]
```diff
@@ -37,70 +26,51 @@ private val xnioSettings = OptionMap.builder()
...
-
-
+val tradeKafkaProducer = fund.cyber.markets.connectors.common.kafka.ConnectorKafkaProducer<Trade>()
+val orderKafkaProducer = fund.cyber.markets.connectors.common.kafka.ConnectorKafkaProducer<OrdersBatch>()
 fun main(args: Array<String>) {
     val debugMode = System.getProperty("debug") != null
 
     supportedTradesEndpoints.forEach { exchange ->
-        concurrent {
+        fund.cyber.markets.connectors.helpers.concurrent {
```
Comment (hleb-albau) [inline connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt]: import?

File: connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt [hunk 8]
```diff
@@ -37,70 +26,51 @@ private val xnioSettings = OptionMap.builder()
...
-val tradeKafkaProducer = ConnectorKafkaProducer<Trade>()
-val orderKafkaProducer = ConnectorKafkaProducer<OrdersBatch>()
-
-
+val tradeKafkaProducer = fund.cyber.markets.connectors.common.kafka.ConnectorKafkaProducer<Trade>()
+val orderKafkaProducer = fund.cyber.markets.connectors.common.kafka.ConnectorKafkaProducer<OrdersBatch>()
 fun main(args: Array<String>) {
     val debugMode = System.getProperty("debug") != null
 
     supportedTradesEndpoints.forEach { exchange ->
```
Comment (hleb-albau) [inline connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt]: do we still need this list?

File: connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt [hunk 9]
```diff
@@ -37,70 +26,51 @@ private val xnioSettings = OptionMap.builder()
...
-        concurrent {
+        fund.cyber.markets.connectors.helpers.concurrent {
             val dataChannel = exchange.subscribe()
-            concurrent {
+            fund.cyber.markets.connectors.helpers.concurrent {
                 while (true) {
                     val message = dataChannel.receive()
                     if (debugMode) println(message)
-                    else orderKafkaProducer.send(OrdersUpdateProducerRecord(message.ordersBatch))
+                    else orderKafkaProducer.send(fund.cyber.markets.connectors.common.kafka.OrdersUpdateProducerRecord(message.ordersBatch))
```
Comment (hleb-albau) [inline connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt]: import

File: devops/prod/docker-compose.yml [hunk 10]
```diff
@@ -48,5 +48,5 @@ services:
     links:
       - cn-kafka
     labels:
-      kompose.service.type: loadbalancer
+      kompose.service.type: loadbalancercd
```
Comment (hleb-albau) [inline devops/prod/docker-compose.yml]: typo

File: build.gradle (hunk 1 above)
Comment (mgergalov) [reply build.gradle]: fixed

File: build.gradle (hunk 2 above)
Comment (mgergalov) [reply build.gradle]: fixed

File: connectors-bitstamp/src/main/kotlin/fund/cyber/markets/connectors/bitstamp/BitstampConnectorApplication.kt (hunk 3 above)
Comment (mgergalov) [reply connectors-bitstamp/src/main/kotlin/fund/cyber/markets/connectors/bitstamp/BitstampConnectorApplication.kt]: fixed

File: connectors-core/src/test/kotlin/fund/cyber/markets/exchanges/hitbtc/HitBtcMessageParserTest.kt (hunk 4 above)
Comment (mgergalov) [reply connectors-core/src/test/kotlin/fund/cyber/markets/exchanges/hitbtc/HitBtcMessageParserTest.kt]: fixed

File: connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt (hunk 5 above)
Comment (mgergalov) [reply connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt]: fixed

File: connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt (hunk 7 above)
Comment (mgergalov) [reply connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt]: fixed

File: connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt (hunk 9 above)
Comment (mgergalov) [reply connectors-hitbtc/src/main/kotlin/fund/cyber/markets/connectors/etherdelta/HitBtcConnectorApplication.kt]: fixed

File: devops/prod/docker-compose.yml (hunk 10 above)
Comment (mgergalov) [reply devops/prod/docker-compose.yml]: fixed

**Pull request merged**